File: day1/haskell/part2.hs
module Main where

main :: IO ()
main = do
    depths <- map (\x -> read x :: Integer) <$> (lines <$> readFile "../input.txt")
    let blocks = map sum (chunkList depths)
    print $ length $ [x | x <- zipWith (<) blocks (drop 1 blocks), x]

chunkList [] = []
chunkList list = take 3 list : chunkList (drop 1 list)

File: day1/python/part2.py
with open("../input.txt") as file:
    depths = list(map(int, file.read().splitlines()));

count = 0
for i in range (0, len(depths)-3):
    if depths[i] + depths[i+1] + depths[i+2] < depths[i+1] + depths[i+2] + depths[i+3]:
        count += 1

print(count)